### PR TITLE
RBF our CPFP transactions

### PR DESCRIPTION
**Pull request by darosior**

### CODE REVIEW

**Comment by darosior** [comment]
I think we should rework `cpfp_txin` on revault_tx side. We probably need to explicit which output is the CPFP, which we could do with a proprietary field.

**Comment by darosior** [comment]
Rebased, just waiting on https://github.com/revault/revault_tx/pull/119 to be merged to point at revault_tx's master

**Review by danielabrozzoni — COMMENTED** [review]
I still need to run the tests locally, but the code looks good :)

File: src/daemon/database/actions.rs [hunk 1]
```diff
@@ -1618,7 +1618,9 @@ mod test {
             ))
         );
         // Not in the CPFPable as it's not broadcasted
-        assert!(!db_cpfpable_spends(&db_path).unwrap().contains(&spend_tx));
+        assert!(!db_cpfpable_spends(&db_path)
+            .unwrap()
+            .contains(&spend_tx.clone()));
```
[the remark this answers is not in the document]
Comment (darosior) [reply src/daemon/database/actions.rs:1623]: A leftover from a previous version

File: src/daemon/bitcoind/poller.rs [hunk 2]
```diff
@@ -441,67 +445,41 @@ impl ToBeCpfped {
...
-    let mut txins = Vec::with_capacity(tx_package.len());
-    for (i, tx) in tx_package.into_iter().enumerate() {
-        let derived_cpfp_descriptor = revaultd
-            .derived_cpfp_descriptor(my_listunspent[i].derivation_index.expect("Must be here"));
-        match tx.cpfp_txin(&derived_cpfp_descriptor) {
+        match tx.cpfp_txin(cpfp_descriptor, &revaultd.secp_ctx) {
             Some(txin) => txins.push(txin),
             None => {
                 log::error!("No CPFP txin for tx '{}'", tx.txid());
                 return Ok(());
```
[the remark this answers is not in the document]
Comment (darosior) [reply src/daemon/bitcoind/poller.rs:468]: Yeah i noted that but didn't want to address it here (it'd also (for now) not happen with a transaction we'd have created).

File: src/daemon/database/actions.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (darosior) [reply src/daemon/database/actions.rs]: Addressed

**Comment by danielabrozzoni** [comment]
ACK ae055f23dc87416d788f33dc303cdfa4c5c70872

**Pull request merged**